packages/analyticsdx-template-lint/src/schemas/readiness-schema.json

```diff
@@ -167,11 +178,12 @@
             "sobject": {
               "type": "string",
               "description": "The name of the Sobject. This can be a ${...} expression.",
+              "minLength": 1,
```

Comment:
Technically, you can deploy a readiness file with empty string for the (s)object/dataset value but it will fail at runtime, so let's warn at edit time.